### DISCUSSION TITLE
How install with customizition ?

### DISCUSSION THREAD

**Discussion by farshidpg** [beyondcode/laravel-websockets#1085] [Q&A]
Hello , I installed 2.X beta ( latest ) for Laravel 9  then I modify and changed  7 files of  this package for add the Webhook events .
Now my changes finish and Webhook  added and work success and  package work.
But  there is a problem, I  edit files from vendor folder directly and I know it's bad and wrong way to edit vendor files.
So I need help for it , for example how can I do  :
1: Create package folder inside my project ( OK )

2: When I run the composer install command  "beyondcode/laravel-websockets"  installing from my local directory  inside package folder  or etc ... or even push my customized version to my git repo and install from my git ... 
could you please help me about this ?

thank you

**Comment by LAGGOUNE-Walid** [comment]
Hello, you can move your modified version (folder in this case) out from vendor directory to laravel app directory.
after that you need to modify composer.json file . this is an example (i modified daniel-de-wit/lighthouse-sanctum): 
```javascript
"autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/",
            "DanielDeWit\\LighthouseSanctum\\" : "app/GraphQL/daniel-de-wit/lighthouse-sanctum/src",
            "Grimzy\\LaravelMysqlSpatial\\": "app//LaravelMysqlSpatial/src/",
            "chillerlan\\QRCode\\" : "chillerlan/php-qrcode/src",
            "chillerlan\\Settings\\" : "chillerlan/php-settings-container/src"
        }
    },
```
